packages/engine-test-utils/src/__tests__/engine-server/EngineEvents.spec.ts

```diff
@@ -0,0 +1,404 @@
+import { NoteChangeEntry, NoteChangeUpdateEntry } from "@dendronhq/common-all";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+import { DendronEngineClient } from "@dendronhq/engine-server";
+import { createEngineFromServer, ENGINE_HOOKS, runEngineTestV5 } from "../..";
+
+/**
+ * Tests that the EngineEvents interface of the client-side DendronEngineClient signals properly
+ */
+describe("GIVEN a DendronEngineClient running on client-side", () => {
+  describe("WHEN writing a new note", () => {
+    test("THEN create event fired with the correct props AND update note event fired for its parent", async (done) => {
+      await runEngineTestV5(
+        async ({ engine, vaults, wsRoot }) => {
+          // TODO: Get rid of this casting once the proper way to expose EngineEvents of DendronEngineClient is implemented
+          const engineClient = engine as DendronEngineClient;
```

Comment:
see earlier comments. if engine can return an event emitter, then we would not need to typecast

Reply:
This won't help though unless we also add `getEngineEventEmitter()` into `DEngineClient`, which I didn't want to do.  A DEngineClient does not necessarily need to also be the `EventsEngineEmitter`; the emitter data may come from elsewhere.

Reply:
But if we do that static method, then we'll just move the casting call to inside that `EngineEventsEmitter.create()` method, which will be worse - I don't think it's that bad to have a cast here, since we're in test code.  If the cast is invalid, the test will fail, which is what we want.  If we move the cast to product code, then if that cast fails we're in a much worse spot.

The proper fix here is to expose EngineEventEmitter to be one of the arg options inside the `runEngineTestV*`, i.e.

```typescript
await runEngineTestV5(
    async ({ engine, vaults, wsRoot, eventEmitter }) => {
    ...
```

But that's a bit more involved, which is why I left it as a `TODO` comment.